src/ducks/context/SelectionContext.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {
+  createContext,
+  useState,
+  useCallback,
+  useEffect,
+  useContext
+} from 'react'
+
+import flag from 'cozy-flags'
+
+export const SelectionContext = createContext()
+
+export const useSelectionContext = () => {
+  return useContext(SelectionContext)
+}
+
+const SelectionProvider = ({ children }) => {
+  const [isSelectionModeActive, setIsSelectionModeActive] = useState(false)
+  const [selected, setSelected] = useState([])
+  const isSelectionModeEnabled = flag('banks.selectionMode.enabled')
+
+  const activateSelectionMode = useCallback(
+    () => isSelectionModeEnabled && setIsSelectionModeActive(true),
+    [isSelectionModeEnabled]
+  )
+
+  const deactivateSelectionMode = () => setIsSelectionModeActive(false)
+
+  const addToSelection = item => {
+    setSelected(v => [...v, item])
+  }
+
+  const removeFromSelection = useCallback(
+    item => {
+      setSelected(selected.filter(e => e.id !== item.id))
+    },
+    [selected]
+  )
+
+  const emptySelection = () => setSelected([])
```

Comment:
useCallback can be used here

Reply:
J'y avais pensé, mais est-ce utile de useCallbacker un fonction qui ne fait que retourner un setter de state ? D'ailleurs est-ce utile de useCallbaker une fonction qui n'utilise aucune variable en paramètre ? 
setSelected ne déclenche pas de re-render, il ne peut pas changer.

Avec un `const emptySelection = useCallback(() => setSelected([]), [])` le linter ne me dit pas qu'il manque une dépendance. Mais est-ce utile d'utiliser un tableau de dépendance vide ?

lien vers la doc https://fr.reactjs.org/docs/hooks-reference.html#usecallback

Reply:
Ici emptySelection est recréée à chaque render et est utilisé dans le contexte. Si on ne memoize pas emptySelection, alors emptySelection en dependance du useMemo qui pourrait être utilisé sur la valeur du contexte changerait tout le temps, et donc le useMemo ne servirait à rien.

Tu peux ignorer le warning du linter ou mettre setSelected en dependance, tout en sachant que ca ne changera pas, donc emptySelection ne changera pas donc la valeur du contexte pourra être correctement memoizée.

Reply:
Le but de useCallback, useMemo c'est de garder l'identité des handlers et des valeurs pour permettre des checks d'egalité par react lorsqu il verifie s'il doit re-renderer.

Reply:
Une autre approche serait de créer toutes les valeurs utilisés dans le contexte dans le useMemo qui wrapperait le contexte.

```
const contextValue = useMemo(() => ({
  emptySelection: () => setSelected([])
}), [setSelected])
```

Reply:
ok top merci pour les précisions 👍